File: app/src/main/java/com/softbankrobotics/qisdktutorials/ui/tutorialtoolbar/TutorialToolbar.java
package com.softbankrobotics.qisdktutorials.ui.tutorialtoolbar;

import android.content.Context;
import android.support.annotation.StringRes;
import android.support.v4.content.ContextCompat;
import android.support.v7.widget.Toolbar;
import android.util.AttributeSet;
import android.view.View;
import android.widget.ImageView;
import android.widget.TextView;

import com.softbankrobotics.qisdktutorials.R;
import com.softbankrobotics.qisdktutorials.model.data.TutorialLevel;

public class TutorialToolbar extends Toolbar {

    private ImageView backArrow;
    private TextView titleTextView;
    private TextView levelTextView;
    private View backgroundView;

    public TutorialToolbar(Context context) {
        this(context, null);
    }

    public TutorialToolbar(Context context, AttributeSet attrs) {
        this(context, attrs, 0);
    }

    public TutorialToolbar(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
        init();
    }

    @Override
    public void setNavigationOnClickListener(OnClickListener listener) {
        backArrow.setOnClickListener(listener);
    }

    public void setName(String name) {
        titleTextView.setText(name);
        invalidate();
        requestLayout();
    }

    public void setName(@StringRes int resId) {
        titleTextView.setText(resId);
        invalidate();
        requestLayout();
    }

    public void setLevel(TutorialLevel level) {
        switch (level) {
            case BASICS:
                levelTextView.setText(R.string.toolbar_basic_level);
                backgroundView.setBackgroundColor(ContextCompat.getColor(getContext(), R.color.basic_green));
                break;
            case ADVANCED:
                levelTextView.setText(R.string.toolbar_advanced_level);
                backgroundView.setBackgroundColor(ContextCompat.getColor(getContext(), R.color.advanced_orange));
                break;
            default:
                throw new IllegalArgumentException("Unknown tutorial level: " + level);
        }

        invalidate();
        requestLayout();
    }

    private void init() {
        inflate(getContext(), R.layout.tutorial_toolbar, this);
        backArrow = findViewById(R.id.backArrow);
        titleTextView = findViewById(R.id.titleTextView);
        levelTextView = findViewById(R.id.levelTextView);
        backgroundView = findViewById(R.id.backgroundView);
    }
}

File: app/src/main/java/com/softbankrobotics/qisdktutorials/ui/tutorialtoolbar/TutorialToolbar.kt
/*
 * Copyright (C) 2018 Softbank Robotics Europe
 * See COPYING for the license
 */

package com.softbankrobotics.qisdktutorials.ui.tutorialtoolbar

import android.content.Context
import androidx.annotation.StringRes
import androidx.core.content.ContextCompat
import androidx.appcompat.widget.Toolbar
import android.util.AttributeSet
import android.view.View

import com.softbankrobotics.qisdktutorials.R
import com.softbankrobotics.qisdktutorials.model.data.TutorialLevel
import kotlinx.android.synthetic.main.tutorial_toolbar.view.*

class TutorialToolbar (context: Context, attrs: AttributeSet? = null, defStyleAttr: Int = 0) : Toolbar(context, attrs, defStyleAttr) {

    init {
        View.inflate(context, R.layout.tutorial_toolbar, this)
    }

    constructor(context: Context): this(context, null)

    constructor(context: Context, attrs: AttributeSet?): this(context, attrs,0)

    override fun setNavigationOnClickListener(listener: OnClickListener) {
        back_arrow.setOnClickListener(listener)
    }

    fun setName(name: String) {
        title_textview.text = name
        invalidate()
        requestLayout()
    }

    fun setName(@StringRes resId: Int) {
        title_textview.text = resources.getString((resId))
        invalidate()
        requestLayout()
    }

    fun setLevel(level: TutorialLevel) {
        when (level) {
            TutorialLevel.BASIC -> {
                level_textview.setText(R.string.toolbar_basic_level)
                background_view.setBackgroundColor(ContextCompat.getColor(context, R.color.basic_green))
            }
            TutorialLevel.ADVANCED -> {
                level_textview.setText(R.string.toolbar_advanced_level)
                background_view.setBackgroundColor(ContextCompat.getColor(context, R.color.advanced_orange))
            }
        }

        invalidate()
        requestLayout()
    }
}
